http-client/src/main/java/com/facebook/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -613,7 +614,7 @@ private HttpRequest buildJettyRequest(Request finalRequest, JettyRequestListener
         if (bodyGenerator != null) {
             if (bodyGenerator instanceof StaticBodyGenerator) {
                 StaticBodyGenerator staticBodyGenerator = (StaticBodyGenerator) bodyGenerator;
-                jettyRequest.content(new BytesContentProvider(staticBodyGenerator.getBody()));
+                jettyRequest.content(new ChunkedBytesContentProvider(staticBodyGenerator.getBody(), requestBufferSizeInBytes));
```

Comment:
hmm... so we want to make `ChunkedBytesContentProvider` to be the default one -- is it always better than the old one?

Reply:
The default one is not safe. It may use arbitrary large native memory buffers. Thus i would argue that the new one is always better. 

The `FileBodyGenerator` uses the same strategy as the `ChunkedBytesContentProvider`